### PR TITLE
EMR on EKS with Karpenter example with V5 EKS Blueprints

### PR DESCRIPTION
**Pull request by vara-bonthu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

- This PR is updated to use latest direction from V5 EKS Blueprints
- EKS Cluster and Node groups will be created using terraform-aws-eks module
- No changes to the EKS Addons for now
-

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/awslabs/data-on-eks/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR to add a new Add-on for [Terraform EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) repo (if applicable)
- [x] Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [x] E2E Test successfully complete before merge?

### CODE REVIEW

File: analytics/terraform/emr-eks-karpenter/emr-eks.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+module "emr_containers" {
+  source = "./modules/emr-eks-containers"
+
+  enable_emr_on_eks     = true
```
Comment (season1946) [inline analytics/terraform/emr-eks-karpenter/emr-eks.tf:5]: Do we still need "enable_emr_on_eks" after separating it out?

File: analytics/terraform/emr-eks-karpenter/addons.tf [hunk 2]
```diff
@@ -1,10 +1,15 @@
+
 module "eks_blueprints_kubernetes_addons" {
-  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.19.0"
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.25.0"
```
Comment (season1946) [inline analytics/terraform/emr-eks-karpenter/addons.tf:3]: what was the decision on add-on module? I thought we gonna use a new repo just for add-on?

File: analytics/terraform/emr-eks-karpenter/addons.tf [hunk 3]
```diff
@@ -15,20 +20,8 @@ module "eks_blueprints_kubernetes_addons" {
   enable_amazon_eks_aws_ebs_csi_driver = true
 
   #---------------------------------------
-  # Metrics Server
+  # Kubernetes Add-ons
   #---------------------------------------
-  enable_metrics_server = true
```
Comment (season1946) [inline analytics/terraform/emr-eks-karpenter/addons.tf:20]: Should we keep metrics server? Prometheus cannot collect kube-state-metrics without metrics server.

**Review by season1946 — APPROVED** [review]
looks good to me. left some open discussion questions. pls review before merging.

File: analytics/terraform/emr-eks-karpenter/addons.tf (hunk 3 above)
Comment (vara-bonthu) [reply analytics/terraform/emr-eks-karpenter/addons.tf:20]: Good point. Should we build kube-state-metrics instead? Fee customers are using this with Prometheus Adaptor

File: analytics/terraform/emr-eks-karpenter/addons.tf (hunk 2 above)
Comment (vara-bonthu) [reply analytics/terraform/emr-eks-karpenter/addons.tf:3]: The source will be changed once the add-ons repo is published under aws-ia org. No change to the code. Our add-ons modules do a lot of heavy lifting so lets keep it this way.

File: analytics/terraform/emr-eks-karpenter/emr-eks.tf (hunk 1 above)
Comment (vara-bonthu) [reply analytics/terraform/emr-eks-karpenter/emr-eks.tf:5]: We can certainly take that out. I will take a look. This sub-module is going to be new repo.

**Pull request merged**